image/oci/layout/oci_dest_test.go

```diff
@@ -216,3 +217,136 @@ func TestPutblobFromLocalFile(t *testing.T) {
+	require.NoError(t, err)
+	require.NotNil(t, desc)
+
+	sigstoreSign := signature.SigstoreFromComponents(
+		"application/vnd.dev.cosign.simplesigning.v1+json",
+		[]byte("test-payload"),
+		map[string]string{"dev.cosignproject.cosign/signature": "test-signature"},
+	)
+
+	err = ociDest.PutSignaturesWithFormat(context.Background(), []signature.Signature{sigstoreSign}, &desc.Digest)
+	require.NoError(t, err)
+
+	err = ociDest.Commit(context.Background(), nil)
+	require.NoError(t, err)
+
+	src, err := ref.NewImageSource(context.Background(), nil)
+	require.NoError(t, err)
+	ociSrc, ok := src.(*ociImageSource)
+	require.True(t, ok)
+	sign, err := ociSrc.GetSignaturesWithFormat(context.Background(), &desc.Digest)
+	require.NoError(t, err)
+	require.Len(t, sign, 1)
+	require.Equal(t, sigstoreSign, sign[0])
+}
+
+// TestPutSignaturesWithFormatTwice tests PutSignaturesWithFormat twice and checks
+func TestPutSignaturesWithFormatTwice(t *testing.T) {
+	tmpDir := loadFixture(t, "single_image_layout")
+	ref, err := NewReference(tmpDir, "latest")
+	require.NoError(t, err)
+	dest, err := ref.NewImageDestination(context.Background(), nil)
+	require.NoError(t, err)
+	defer dest.Close()
+	ociDest, ok := dest.(*ociImageDestination)
+	require.True(t, ok)
+
+	desc, _, err := ociDest.ref.getManifestDescriptor()
+	require.NoError(t, err)
+	require.NotNil(t, desc)
+
+	sigstoreSign := signature.SigstoreFromComponents(
+		"application/vnd.dev.cosign.simplesigning.v1+json",
+		[]byte("test-payload"),
+		map[string]string{"dev.cosignproject.cosign/signature": "test-signature"},
+	)
+	sigstoreSign2 := signature.SigstoreFromComponents(
+		"application/vnd.dev.cosign.simplesigning.v1+json",
+		[]byte("test-payload2"),
+		map[string]string{"dev.cosignproject.cosign/signature": "test-signature"},
+	)
+
+	err = ociDest.PutSignaturesWithFormat(context.Background(), []signature.Signature{sigstoreSign}, &desc.Digest)
+	require.NoError(t, err)
+
+	err = ociDest.Commit(context.Background(), nil)
+	require.NoError(t, err)
+
+	err = ociDest.PutSignaturesWithFormat(context.Background(), []signature.Signature{sigstoreSign, sigstoreSign2}, &desc.Digest)
+	require.NoError(t, err)
+
+	err = ociDest.Commit(context.Background(), nil)
+	require.NoError(t, err)
+
+	src, err := ref.NewImageSource(context.Background(), nil)
+	require.NoError(t, err)
+	ociSrc, ok := src.(*ociImageSource)
+	require.True(t, ok)
+	sign, err := ociSrc.GetSignaturesWithFormat(context.Background(), &desc.Digest)
+	require.NoError(t, err)
+	require.Len(t, sign, 2)
+	require.Equal(t, sigstoreSign, sign[0])
+	require.Equal(t, sigstoreSign2, sign[1])
+}
+
+// TestPutSignaturesWithFormatNilDigest tests error handling when instanceDigest is nil
+func TestPutSignaturesWithFormatNilDigest(t *testing.T) {
+	ref, _ := refToTempOCI(t, false)
+
+	dest, err := ref.NewImageDestination(context.Background(), nil)
+	require.NoError(t, err)
+	defer dest.Close()
+
+	// Cast to ociImageDestination to access PutSignaturesWithFormat
```

Comment:
Using `imagedestination.FromPublic` would be more similar to actual callers, and thus more representative. (Also elsewhere.)